packages/core/util/index.ts

```diff
@@ -865,3 +865,118 @@ export const rIC =
   (typeof jest === 'undefined'
     ? (cb: Function) => setTimeout(() => cb(), 1)
     : (cb: Function) => cb())
+
+// xref https://gist.github.com/tophtucker/62f93a4658387bb61e4510c37e2e97cf
+export function measureText(str: string, fontSize = 10) {
+  // prettier-ignore
+  const widths = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.2796875,0.2765625,0.3546875,0.5546875,0.5546875,0.8890625,0.665625,0.190625,0.3328125,0.3328125,0.3890625,0.5828125,0.2765625,0.3328125,0.2765625,0.3015625,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.5546875,0.2765625,0.2765625,0.584375,0.5828125,0.584375,0.5546875,1.0140625,0.665625,0.665625,0.721875,0.721875,0.665625,0.609375,0.7765625,0.721875,0.2765625,0.5,0.665625,0.5546875,0.8328125,0.721875,0.7765625,0.665625,0.7765625,0.721875,0.665625,0.609375,0.721875,0.665625,0.94375,0.665625,0.665625,0.609375,0.2765625,0.3546875,0.2765625,0.4765625,0.5546875,0.3328125,0.5546875,0.5546875,0.5,0.5546875,0.5546875,0.2765625,0.5546875,0.5546875,0.221875,0.240625,0.5,0.221875,0.8328125,0.5546875,0.5546875,0.5546875,0.5546875,0.3328125,0.5,0.2765625,0.5546875,0.5,0.721875,0.5,0.5,0.5,0.3546875,0.259375,0.353125,0.5890625]
+  const avg = 0.5279276315789471
+  return (
+    str
+      .split('')
+      .map(c =>
+        c.charCodeAt(0) < widths.length ? widths[c.charCodeAt(0)] : avg,
+      )
+      .reduce((cur, acc) => acc + cur) * fontSize
+  )
+}
```

Comment:
if this is being called a lot it might be made faster by avoiding the split/map/reduce and instead just iterating with a for loop

Reply:
it's pretty infrequently used. It is sort of a hack to avoid doing proper width estimation which can be difficult with real element.getBBox()...this could be removed from core if it seems too hacky to have in there